### PR TITLE
Typify tables

### PR DESCRIPTION
**Pull request by vorot93**
This PR changes database driver to accept user-defined types, as specified for each table.

Key highlights:
- Tables along with key/value types are now specified in `kv/tables.rs`, `akula-table-defs` crate has been removed.
- `TableEncode`, `TableDecode`, `TableObject` traits which should be implemented by types that are to be stored in the database.
- `Table` now has `Key`, `Value`, `FusedValue` and `SeekKey` associated types. `FusedValue` is `(Key, Value)` for all tables except `PlainState` which stores multiple value types depending on the key.
- AutoDupSort magic has been removed.

### CODE REVIEW

File: src/accessors/chain.rs [hunk 1]
```diff
@@ -61,88 +41,68 @@ pub mod canonical_hash {
...
-            while let Some((_, tx_rlp)) = walker.try_next().await? {
-                out.push(rlp::decode(&tx_rlp).context("broken tx rlp")?);
-
-                if out.len() >= amount as usize {
-                    break;
-                }
-            }
-
-            out
+            futures_util::TryStreamExt::try_collect(
```
Comment (blasrodri) [inline src/accessors/chain.rs:91]:
```diff
-            futures_util::TryStreamExt::try_collect(
-                walker.take(amount).map(|res| res.map(|(_, v)| v)),
-            )
+                walker.take(amount).map(|res| res.map(|(_, v)| v)).try_collect()
```

File: src/accessors/chain.rs [hunk 2]
```diff
@@ -205,71 +169,51 @@ pub mod tx_sender {
...
 
-        for (i, sender) in senders.iter().enumerate() {
-            let key = (base_tx_id + i as u64).to_be_bytes();
-            let data = sender.to_fixed_bytes();
-            cursor.put(&key, &data).await.unwrap();
+        for (i, &sender) in senders.iter().enumerate() {
+            cursor
+                .put((TxIndex(base_tx_id.0 + i as u64), sender))
+                .await
+                .unwrap();
```
Comment (blasrodri) [inline src/accessors/chain.rs:176]: Is there anything useful we could report here in case of a panic?

File: src/changeset/mod.rs [hunk 3]
```diff
@@ -1,7 +1,10 @@
-use crate::{kv::*, models::*, CursorDupSort, *};
-use arrayref::array_ref;
+use crate::{
+    kv::{tables::BitmapKey, *},
+    models::*,
+    CursorDupSort,
+};
 use async_trait::async_trait;
-use bytes::Bytes;
+use roaring::RoaringTreemap;
```
Comment (blasrodri) [inline src/changeset/mod.rs:7]: what's the benefit of changing these two?

**Review by blasrodri — APPROVED** [review]
Left minor comments. Great upgrade to the codebase. Awesome work!

File: src/changeset/mod.rs (hunk 3 above)
Comment (vorot93) [reply src/changeset/mod.rs:7]: RoaringTreemap is now part of table definition and parsed in database layer, not in changesets.

**Pull request merged**